src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -41,6 +41,20 @@
 #define TRANSIENT_FAILURE_ATTEMPTS 3 /* Arbitrary limit: how many attempts are considered enough to report
                                       * transient failure. */
 
+/* RFC2131 section 4.1 specifies retransmission delays, starting at 4s then doubling
+ * each attempt with max of 64s, with -1 to +1 sec of random 'fuzz' added */
+#define RFC2131_FUZZ_USEC ((int32_t)(random_u32() % (2 * USEC_PER_SEC)) - (int32_t)USEC_PER_SEC)
```

Comment:
so, the type is weird here, still. I understand you want signed values here, but why 32bit? usec_t is 64bit, and USEC_PER_SEC is 64bit too, like all µs values we deal with. Thus it would appear a lot more obvious to me if you'd use int64_t here, and random_u64()... i.e. suddenly mixing in a 32bit value when our usual time operations are all 64bit is irritating at best, but smells like an error at worst

Reply:
yep you're right, it should be int64_t; i changed the fuzz to use that type, and also changed to use the usec_add(), usec_sub_unsigned(), and usec_sub_signed() functions, which nicely handle some of the underflow/overflow details, and hopefully make the macros clearer.